src/virtq.c

```diff
@@ -0,0 +1,66 @@
+#include <linux/kvm.h>
+#include <sys/ioctl.h>
+#include <unistd.h>
+
+#include "virtq.h"
+
+void virtq_complete_request(struct virtq *vq)
+{
+    vq->ops->complete_request(vq);
+}
+
+void virtq_notify_used(struct virtq *vq)
+{
+    vq->ops->notify_used(vq);
+}
+
+void virtq_enable(struct virtq *vq)
+{
+    vq->ops->enable_vq(vq);
+}
+
+void virtq_disable(struct virtq *vq) {}
+
+#define VIRTQ_SIZE 128
+void virtq_init(struct virtq *vq, void *dev, struct virtq_ops *ops)
+{
+    vq->info.size = VIRTQ_SIZE;
```

Comment:
Why not take this value from `(struct virtio_pci_dev *) dev->config.common_cfg.queue_size` but using a hard code value instead?

Reply:
Sorry for I might get it wrong, so will this field get modified during the execution the Linux in somewhere like `virtio_pci_space_write`?

Reply:
The size of a queue will not change, but the `queue_size` will be modified in `virtio_pci_select_virtq()`.

Reply:
I see. Sorry for misunderstanding how we sync the queue size between Os and virtio. 

No problem for this question now.